### PR TITLE
Setup property tests, add codec test for MongoDB mounts

### PR DESCRIPTION
**Pull request by garyb**

### CODE REVIEW

File: test/src/Test/Property/Mount/MongoDB.purs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+newtype TestConfig = TestConfig MDB.Config
+
+derive instance eqTestConfig ∷ Eq TestConfig
+
+instance showTestConfig ∷ Show TestConfig where
+  show (TestConfig cfg) = show (MDB.toJSON cfg)
+
+check ∷ ∀ eff. SC eff Unit
+check = (quickCheck :: Gen Result -> SC eff Unit) do
```
Comment (natefaubion) [inline test/src/Test/Property/Mount/MongoDB.purs]: Why this annotation? Also, not unicode.

File: src/Quasar/Mount/Common/Gen.purs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+genPort = Gen.chooseInt 50000 65535
+
+genHost ∷ forall m. MonadGen m ⇒ MonadRec m ⇒ m MDB.Host
+genHost = Tuple <$> genHostURI <*> GenC.genMaybe genPort
+
+genDirPath ∷ forall m. MonadGen m ⇒ MonadRec m ⇒ m DirPath
+genDirPath = Gen.sized \size → do
+  newSize <- Gen.chooseInt 0 size
+  Gen.resize (const newSize) do
+    parts :: L.List String ← Gen.unfoldable genAlphaNumericString
```
Comment (natefaubion) [inline src/Quasar/Mount/Common/Gen.purs]: Not unicode.

**Comment by garyb** [comment]
Fixed the unicode issues. As for the annotation... it doesn't typecheck without it 😛

**Comment by natefaubion** [comment]
I meant more the way it's annotated (why annotate `quickCheck`). Just for aesthetics?

**Comment by garyb** [comment]
Oh I see, yeah that is kinda weird... I didn't really think about it. I'll move the `Gen` into a where and annotate that instead.

File: test/src/Test/Property/Mount/MongoDB.purs [hunk 3]
```diff
@@ -32,7 +32,10 @@ instance showTestConfig ∷ Show TestConfig where
   show (TestConfig cfg) = show (MDB.toJSON cfg)
 
 check ∷ ∀ eff. SC eff Unit
-check = (quickCheck ∷ Gen Result → SC eff Unit) do
-  configIn ← genConfig
-  let configOut = MDB.fromJSON (MDB.toJSON configIn)
-  pure $ Right (TestConfig configIn) === TestConfig <$> configOut
+check = quickCheck prop
+  where
+  prop :: Gen Result
```
Comment (natefaubion) [inline test/src/Test/Property/Mount/MongoDB.purs]: Unicode.... 😆

**Comment by garyb** [comment]
🙈... fixed!

**Review by natefaubion — APPROVED** [review]

**Pull request merged**